src/loki/search.cc

```diff
@@ -458,11 +477,23 @@ struct bin_handler_t {
           continue;
       }
 
-      // reset these so we know the best point along the edge
+      // initialize candidates vector:
+      // - reset sq_distance to max so we know the best point along the edge
+      // - apply prefilters based on user's SearchFilter request options
       auto c_itr = bin_candidates.begin();
       decltype(begin) p_itr;
+      bool all_prefiltered = true;
       for (p_itr = begin; p_itr != end; ++p_itr, ++c_itr) {
         c_itr->sq_distance = std::numeric_limits<float>::max();
+        c_itr->prefiltered = is_search_filter_triggered(edge, p_itr->location.search_filter_);
+        if (!c_itr->prefiltered) {
```

Comment:
i dont think we need a branch here. cant we just do `all_prefiltered = all_prefiltered && c_itr->prefiltered`. so that any time one is `false` it stays `false`. save those clock cycles!